### PR TITLE
Set up a "Hello world" project with two apps

### PR DESCRIPTION
**Pull request by mrkamin**
- [ ] Project requirements

- Create a new React app called 'hello-react-front-end'
- Initialize your project with Git.
- Create the App component with react-router.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- Display the random greeting in your Greeting component.
<summary>Backend Api Link</summary>
  <ul>
    <li><a href="http://localhost:3000/api/greeting">Rails as Backend - API's used</a></li>
  </ul>
</details>
<summary>Backend PR Link</summary>
  <ul>
    <li><a href="https://github.com/mrkamin/Hello-Rails-Back-End/pull/1">Rails as Backend - PR Link</a></li>
  </ul>
</details>

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @mrkamin, 


descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't use the correct link for the PR.

When submitting for a review - **use a link to the pull request in your Rails API.**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+
+    <Router>
+      <div className="App">
+        <Greeting />
+        <Routes />
```
Comment (mikethreels) [inline src/App.js]: - [x] it seems you have setup the Routes incorrectly, the route is not leading anywhere and you're only calling the greeting component.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi @mrkamin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job connecting the 2 apps

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,18 @@
...
-        </a>
-      </header>
-    </div>
+    <Provider store={store}>
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<Greeting />} />
+        </Routes>
+      </BrowserRouter>
+    </Provider>
```
Comment (mrkamin) [inline src/App.js:15]: @mikethreels 
thanks you for your good and valuable feedback, 
I apply your comment to the project

File: src/redux/greetings/greetingsSlice.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import axios from 'axios';
+
+const initialState = {
+  greeting: '',
+  error: null,
+};
+
+export const getRandomGreeting = createAsyncThunk('getRandomGreeting', async () => {
+  try {
+    const response = await axios.get('http://localhost:3000/api/greeting');
```
Comment (julie-ify) [inline src/redux/greetings/greetingsSlice.js:11]: - [x] Your frontend does not fetch the greeting from the API due to cross-origin blockage so I cannot see any messages. In your backend, you need to install the `rack-cors` gem and configure the middleware to allow requests from the frontend.

  See this link for setting up CORS in the backend: https://github.com/cyu/rack-cors

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mrkamin,

Good job so far!

## Highlights!
- Good code organization and working apps ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/greetings/greetingsSlice.js (hunk 3 above)
Comment (mrkamin) [reply src/redux/greetings/greetingsSlice.js:11]: @julie-ify,
thanks for your good and valuable feedback, 
the cors setup according to the link you provided.

**Pull request merged**